signature/mechanism_gpgme.go

```diff
@@ -133,12 +134,41 @@ func (m *gpgmeSigningMechanism) Sign(input []byte, keyIdentity string) ([]byte,
 	if err != nil {
 		return nil, err
 	}
+
+	if passphrase != "" {
+		if !strings.HasSuffix(passphrase, "\n") { // passphrase needs to have a trailing new line
+			passphrase += "\n"
```

Comment:
AFAICS  a `'\n'` inside the passphrase is always interpreted as a terminator inside the callback, so I’d rather explicitly reject any such input, instead of silently using a different passphrase value. If we ever implemented a Go-native mechanism, any hypothetical users passing a string with trailing `'\n'` would be broken.